### PR TITLE
SN-1453 bootstrap application

### PR DESCRIPTION
**Pull request by jls-serge**
https://juliusinc.atlassian.net/browse/SN-1453

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: "3.7"
+
+services:
+  youtube-platform:
```
Comment (ivanbochko) [inline docker-compose.yml:4]: youtube artifact to rename :)

File: src/config/app.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-export const port = parseInt(process.env.APP_PORT, 10) || 3000;
```
Comment (ivanbochko) [inline src/config/app.ts:1]: lets change it to 3000 back, 9000 is reserved for platforms usually. Look docker-compose.yml to update port

File: src/config/app.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
...
+export const port = parseInt(process.env.APP_PORT, 10) || 9000;
 export const env = process.env.NODE_ENV || 'dev';
-export const globalPrefix = 'template';
 export const corsExceptions = process.env.CORS_EXCEPTIONS || 'https://influencers.juliusdev.net,http://localhost:4300';
+export const globalPrefix = '';
```
Comment (ivanbochko) [inline src/config/app.ts:4]: lets have global prefix, like `scheduler`

File: src/jobs/BaseJob.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        this.options = options || {}
+    }
+
+    getPlatform() {
+        return this.platform;
+    }
+}
+
+
+export enum PLATFORM {
```
Comment (ivanbochko) [inline src/jobs/BaseJob.ts:19]: Lets name it camelcase Platform maybe?

File: src/jobs/BaseJob.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+export default class BaseJob {
+    protected platform: PLATFORM;
+    protected accounts: number[];
+    protected options: any;
+    protected readonly jobName: string;
```
Comment (ivanbochko) [inline src/jobs/BaseJob.ts:5]: how do you assign jobName?

I'd repeat legacy message structure for compatibility:
```
{
  name: 'job name',
  data: {}
}
```

File: src/prisma/prisma.service.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    async onModuleDestroy() {
+        await this.$disconnect();
+    }
+
+    /**
+     * @param {string} platformName
+     * @param {Object} options
+     */
+    async selectHighPriorityAccounts(platformName, options: any = {}) {
```
Comment (ivanbochko) [inline src/prisma/prisma.service.ts:20]: I'd leave prismaservice to be responsible only about connection and use it in more high level services.

File: src/rabbit/rabbit.wrap.service.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import {ConfigService} from 'nestjs-config';
+import * as amqp from 'amqplib/callback_api';
+import BaseJob from '../jobs/BaseJob';
+
+@Injectable()
+export default class RabbitWrapService {
+    private amqpConnection: any;
+
+    constructor(private readonly configService: ConfigService) {
+        amqp.connect(configService.get('rabbitMqUrl'), (error, connection) => {
```
Comment (ivanbochko) [inline src/rabbit/rabbit.wrap.service.ts:11]: lets connect here like prisma service does - on module init/destroy.

File: src/rabbit/rabbit.wrap.service.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+            this.amqpConnection = connection;
+        });
+
+    }
+
+    /**
+     * @param {BaseJob} job
+     */
+    sendPlatformJob(job: BaseJob): any {
```
Comment (ivanbochko) [inline src/rabbit/rabbit.wrap.service.ts:24]: I'd suggest to make rabbit service only responsible to connect/create channel/send to queue/consume.
Other more high level services will use rabbitservice

**Review by ivanbochko — COMMENTED** [review]
great peace of work. thanks. few comments though

File: src/jobs/BaseJob.ts (hunk 5 above)
Comment (jls-serge) [reply src/jobs/BaseJob.ts:5]: Actually I wasn,t planning to put this part of code to this task and implement completely with Instagram task. 
So let's assume it's a billet? )

**Review by ivanbochko — APPROVED** [review]
gj

**Pull request merged**